src/controls/LevelingSensor/QskLevelingSensor.cpp

```diff
@@ -0,0 +1,175 @@
+#include "QskLevelingSensor.h"
+#include <QskFunctions.h>
+#include <QskScaleTickmarks.h>
+#include <unordered_map>
+
+#include <QVector2D>
+#include <QVector3D>
+
+namespace
+{
+    template< typename T >
+    bool compareExchange( T& dst, const T& src )
+    {
+        if ( dst != src )
+        {
+            dst = src;
+            return true;
+        }
+        return false;
+    }
+
+    template<>
+    bool compareExchange< float >( float& dst, const float& src )
```

Comment:
Does this cover the double case as well?

Reply:
No, since the double case is not being used in this file. Still added it so future changes don't run into subtle narrowing type conversions.